services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/converter/attribute/AttributeConverter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.enhanced.dynamodb.converter.attribute;
+
+import java.time.Instant;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.enhanced.dynamodb.converter.Converter;
+import software.amazon.awssdk.enhanced.dynamodb.converter.attribute.bundled.InstantAsIntegerAttributeConverter;
+import software.amazon.awssdk.enhanced.dynamodb.converter.attribute.bundled.InstantAsStringAttributeConverter;
+import software.amazon.awssdk.enhanced.dynamodb.converter.attribute.bundled.StringAttributeConverter;
+import software.amazon.awssdk.enhanced.dynamodb.model.AttributeConverterAware;
+import software.amazon.awssdk.enhanced.dynamodb.model.ItemAttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+/**
+ * Converts between a specific Java type and an {@link ItemAttributeValue}.
+ *
+ * <p>
+ * More specifically, this converts Java types into {@link ItemAttributeValue}s, which are more user-friendly representations of
+ * the generated {@link AttributeValue}. An {@link ItemAttributeValue} can always be converted to a generated
+ * {@link AttributeValue} using {@link ItemAttributeValue#toGeneratedAttributeValue()}.
+ *
+ * <p>
+ * Examples:
+ * <ul>
+ *     <li>The {@link StringAttributeConverter} converts a {@link String} into a DynamoDB string
+ *     ({@link AttributeValue#s()}).</li>
+ *     <li>The {@link InstantAsIntegerAttributeConverter} converts an {@link Instant} into a DynamoDB number
+ *     ({@link AttributeValue#n()}).</li>
+ *     <li>The {@link InstantAsStringAttributeConverter} converts an {@link Instant} into a DynamoDB string
+ *     ({@link AttributeValue#s()}).</li>
+ * </ul>
+ *
+ * <p>
+ * Unlike {@link SubtypeAttributeConverter}, this does not support subtypes of a specific type.
+ *
+ * <p>
+ * See {@link AttributeConverterAware} for more information on how converters are used by the mapper.
+ *
+ * @param <T> The Java type supported by this converter.
+ *
+ * @see AttributeConverterAware
+ * @see SubtypeAttributeConverter
+ */
+@SdkPublicApi
+@ThreadSafe
+public interface AttributeConverter<T> extends Converter<T> {
+    /**
+     * Convert the provided Java object into an {@link ItemAttributeValue}. This will raise a {@link RuntimeException} if the
```

Comment:
Why `RuntimeException` instead of `SdkClientException`?

Reply:
It's similar to our behavior elsewhere we use the `Validate` class to validate inputs - we raise a `RuntimeException`.